### ISSUE TITLE
lando init with platformsh recipe fails after entering token

### ISSUE THREAD

**Issue by gilzow** [lando/platformsh#184]
Using Lando 3.9.0 on macOS

`lando init` Errors after entering a token (or after selecting an account if PSH token is already set up)

Occurs with either an empty directory, of if running it when the project's code is present in the directory

Prompt for token
![Capture d’écran 2023-01-20 à 15 16 18](https://user-images.githubusercontent.com/2332956/213732679-8ba00954-49ce-41b2-833c-4f067e2269e8.png)

Empty directory, token set
<img width="512" alt="Screen Shot 2023-01-20 at 9 12 10 AM" src="https://user-images.githubusercontent.com/2332956/213732744-a1020dc1-f6eb-4f61-81e3-36724450ecd3.png">

Project code in directory, token set
<img width="526" alt="Screen Shot 2023-01-20 at 9 14 24 AM" src="https://user-images.githubusercontent.com/2332956/213733102-f54a204f-c7e7-4f5f-ab62-45599efe3a8a.png">

Adding `--verbose` flag
<img width="735" alt="Screen Shot 2023-01-20 at 9 16 16 AM" src="https://user-images.githubusercontent.com/2332956/213733525-c3195662-1057-4a95-8ee8-a8127e755761.png">

Happy to provide additional information and/or debug further.

**Comment by gilzow** [comment]
coming back to this as it is still an issue after updating to lando v3.11.0

```
❯ lando version
v3.11.0
```

Docker:
```
❯ docker --version
Docker version 20.10.22, build 3a2c30b
```

Docker desktop v4.16.2

*Any* combination of selecting the platformsh recipe results in 
```
ERROR ==>
```
If I add the `--verbose` flag the error is still the same as originally posted.

**Comment by RightsandWrongsgit** [comment]
Did you get the Docker versions from Lando's 3.11.0 or did you independently update Docker?  I have always found success trailing Docker's updates and letting Lando be the controlling point for the Docker version I run.

**Comment by gilzow** [comment]
I never update Docker outside of lando, letting lando control when I update docker.

**Comment by philippemellenger** [comment]
An issue happened to me as well using platformsh after doing a lando start, I didn't update anything. Lando start was perfectly working last week.
lando destroy && lando rebuild didn't fix the issue. 

`ERROR ==> 2023-02-15 11:39:59,591 root INFO Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt`

Not sure if it is related but since the most recent issue was about platformsh, it might be...

**Comment by philippemellenger** [comment]
An issue happened to me as well using platformsh after doing a lando start, I didn't update anything. Lando start was perfectly working last week.
lando destroy && lando rebuild didn't fix the issue.

ERROR ==> 2023-02-15 11:39:59,591 root INFO Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt

Not sure if it is related but since the most recent issue was about platformsh, it might be...

**Comment by gilzow** [comment]
@pirog @reynoldsalec is there anything I can do to help troubleshoot this issue? We've had a couple of other people report to us they're experiencing the same issue.

**Comment by philippemellenger** [comment]
I think I experienced this after specifying the version of PHP to 8.1 and using the platformsh recipe in the lando.yml
Not sure if it will help you reproduce.

**Comment by gilzow** [comment]
_So far_ existing projects are fine.  It only happens where I'm trying to start from scratch (ie running `lando init`). 
Alternatively, I can add a `lando.yml` file with `name:` and `recipe:platformsh` properties defined and then run `lando start` and everything works. In my experience so far, it's isolated to just `lando init`

**Comment by reynoldsalec** [comment]
Thanks for narrowing that down @gilzow, I'll take a look and replicate later today. Chatting with Chad and Jan tomorrow, so they might be able to answer P.sh-related questions if I have them.

**Comment by reynoldsalec** [comment]
Based on the error I'm seeing when running `lando init -vvv`, my guess is that there's been a change in `platformsh-client`. Specifically I'm seeing...

```
lando 09:34:33 ERROR ==> Error
    at /Users/alec/.lando/plugins/@lando/platformsh/recipes/platformsh/init.js:39:40
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

...which points to an error when we call the `getAccountInfo()` function of `platformsh-client`. Going to see if anything changed about the implementation of that function that would cause an error...hopefully we just need to update our usage of the `platformsh-client`.

**Comment by reynoldsalec** [comment]
I poked around trying different versions of platformsh-client and I do notice a change in the error behavior between version 0.1.198 , where the process just seems to hang when we call `api.getAccountInfo()`, and 0.1.199  which throws an error (sadly an unhelpful error, since platformsh-client  is minified). It looks like `isomorphic-fetch` was updated between those two tags and the function that retrieves the user (which is what `getAccountInfo()` relies upon), so it does seem like a likely place to focus on. Timing also seems to make since, considering the versions of `platformsh-client` the Lando integration has been bundling in its releases.

**Comment by DavidVaini** [comment]
What is the work around for this in the mean time? I started a new project and I cant seem to get lando init working.

**Comment by gilzow** [comment]
@DavidVaini , I don't believe there is one yet. 
If you have an existing project on platform.sh, you can clone it, then add a `.lando.yaml` file with the `name:` property and `recipe: platformsh` and it should work with `lando start` instead of `lando init`.  

Example `.lando.yaml`
```yaml 
name: myapp
recipe: platformsh
```

**Comment by gilzow** [comment]
@reynoldsalec The [platformsh js client has been updated](https://github.com/platformsh/platformsh-client-js/issues/219#issuecomment-1515236573). Can you see if this update resolves this issue? Thanks!

**Comment by reynoldsalec** [comment]
Should have a chance to check it out before the weekend, thanks for the ping @gilzow!

**Comment by reynoldsalec** [comment]
Looks like this did the trick, merged in on PR #188. Note that you'll have to install the [latest version of the P.sh plugin](https://docs.lando.dev/platformsh/getting-started.html#custom-installation) until this change is reflected in the next Lando general release.

Note that I had to refresh my token to get things to work...would be interested if other folks have the same experience.

**Comment by gilzow** [comment]
> Looks like this did the trick, merged in on PR #188. Note that you'll have to install the [latest version of the P.sh plugin](https://docs.lando.dev/platformsh/getting-started.html#custom-installation) until this change is reflected in the Lando CLI 3.16.0 release.

FANTASTIC NEWS!!! Thanks so much! 

> Note that I had to refresh my token to get things to work...would be interested if other folks have the same experience.

I'll try testing it this afternoon and see if I run into the same scenario with the token.

**Comment by gilzow** [comment]
THANK YOU @reynoldsalec !!!!
